sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConstants.java

```diff
@@ -495,7 +499,7 @@ public enum RntbdRequestHeader implements RntbdHeader {
         ConsistencyLevel((short) 0x0010, RntbdTokenType.Byte, false),
         EntityId((short) 0x0011, RntbdTokenType.String, false),
         ResourceSchemaName((short) 0x0012, RntbdTokenType.SmallString, false),
-        ReplicaPath((short) 0x0013, RntbdTokenType.String, true),
+        ReplicaPath((short) 0x0013, RntbdTokenType.String, false), // true in direct, but not for thin client
```

Comment:
This makes it a bit best effort for direct since we don't have a minimal set of headers check for direct in the SDK - our live test pipeline (CI pipeline / Emulator too) can easily catch this though.

Reply:
ACK - but ok, IMO - backend chacks this as well and existing test coverage would be sufficient to find regressions where Java stops sending ReplicaPath header.